agent/consul/intention_endpoint.go

```diff
@@ -841,11 +799,18 @@ func (s *Intention) Check(
 	if err != nil {
```

Comment:
^^- I don't think this matters in practice, but I wanted to mention as a side effect of this change, now we _always_ resolve the anonymous token, instead of just when the intentions didn't match.

Reply:
Hmm actually, we're already resolving the provided token to an authorizer. I removed the anonymous token resolution in favor of the existing authz.

If we reach the state store function that means either:
- authz is nil because ACLs are disabled
- authz is not nil and our token is valid for the intention

If it's not nil, there's going to be a chain of authorizers:
- A policy authorizer
- An enterprise policy authorizer
- A root authorizer based on the ACL defaults

`IntentionDefaultAllow()` can't be determined from the first two and the root authorizer will be the same for the anonymous token.